### ISSUE TITLE
Canonical: Components cannot be fully initialized before `ApplicationRef.tick` with ZoneJS and `TestBed.createComponent`

### ISSUE THREAD

**Issue by atscott** [angular/angular#57858]
### Which @angular/* package(s) are the source of the bug?

core, zone.js

### Is this a regression?

Yes

### Description

TL;DR: Calling `TestBed.createComponent` will cause automatic application synchronization before the tester has an opportunity to fully initialize the component state.

`TestBed.createComponent` wraps the [component creation in an `NgZone.run`](https://github.com/angular/angular/blob/4231e8f28ffe8f55dddc2af0647b5b04fa8445d7/packages/core/testing/src/test_bed.ts#L701). When this closure exits, `NgZone` checks if the zone is stable, and if so, [emits `onMicrotaskEmpty`](https://github.com/angular/angular/blob/4231e8f28ffe8f55dddc2af0647b5b04fa8445d7/packages/core/src/zone/ng_zone.ts#L364-L367). This will cause automatic [application synchronization](https://github.com/angular/angular/blob/4231e8f28ffe8f55dddc2af0647b5b04fa8445d7/packages/core/src/change_detection/scheduling/ng_zone_scheduling.ts#L48-L58) before the tester has a chance to fully initialize the component (i.e. set inputs, run change detection before `afterNextRender` hooks run if they're defined in constructors, etc.). 

One workaround that developers can use today is to wrap all of their setup logic in an `NgZone.run` of their own, i.e.

```
const fixture = TestBed.inject(NgZone).run(() => {
  const fixture = TestBed.createComponent(MyComp);
  fixture.componentRef.setInput('requiredInput', 'initialValue');
  return fixture;
});
```

Related issues:

* https://github.com/angular/angular/issues/57313
* https://github.com/angular/angular/issues/57767
* https://github.com/angular/angular/issues/57856

**Comment by alxhub** [comment]
Fixed by #62040!
